rules/framework.bzl

```diff
@@ -275,7 +267,7 @@ def _get_framework_files(ctx, deps):
             if provider in dep:
                 # collect headers
                 has_header = False
-                for hdr in dep[apple_common.Objc].direct_headers:
+                for hdr in _get_direct_headers(provider, dep):
```

Comment:
Is it possible the semantic of these providers has changed, but the API hasn't? Maybe try something like this:
```
if _detect_old_provider(provider):
     hdrs = _get_hdrs_old()
else
     hdrs = get_hdrs_new()

.. then process hdrs below in the same routine.
```

I can take look at this as well - it should build on a recent cut of rules swift, maybe worth breaking backwards compat if it's too hard

Reply:
Found the problem, it was our use of the `PrivateHeadersInfo` provider.